R/qCount.R

```diff
@@ -3,516 +3,895 @@
+#'   value of \code{reportLevel}. If \code{reportLevel="exon"},
+#'   alignments overlapping each exon in \code{query} are counted.
+#'   If \code{reportLevel="gene"}, alignment counts for all exons of a
+#'   gene will be summed, counting each alignment only once even if it
+#'   overlaps multiple annotated exons of a gene. These are useful to
+#'   calculate exon or gene expression levels in RNA-seq experiments
+#'   based on the annotation in a \code{TxDb} object. If
+#'   \code{reportLevel="promoter"}, the \code{promoters} function from package
+#'   \pkg{GenomicFeatures} is used with default arguments to extract
+#'   promoter regions around transcript start sites, e.g. to quantify
+#'   alignments inf a ChIP-seq experiment.
+#'   \item any of the above or \code{NULL} for
+#'   \code{reportLevel="junction"}: The \code{query} argument is ignored
+#'   if \code{reportLevel} is set to \code{"junction"}, and \code{qCount}
+#'   will count the number of alignments supporting each exon-exon
+#'   junction detected in any of the samples in \code{proj}. The
+#'   arguments \code{selectReadPosition}, \code{shift},
+#'   \code{orientation}, \code{useRead} and \code{mask} will have no
+#'   effect in this quantification mode.
+#' }
+#' 
+#' The additional arguments allow to fine-tune the quantification:
+#' 
+#' \code{selectReadPosition} defines the part of the alignment that has
+#' to be contained within a query region for an overlap. The values
+#' \code{start} (default) and \code{end} refer to the biological start
+#' (5'-end) and end (3'-end) of the alignment. For example, the
+#' \code{start} of an alignment on the plus strand is its leftmost
+#' (lowest) base, and the \code{end} of an alignment on the minus strand
+#' is also the leftmost base.
+#' 
+#' \code{shift} allows on-the-fly shifting of alignments towards their
+#' 3'-end prior to overlap determination and counting. This can be
+#' helpful to increase resolution of ChIP-seq experiments by moving
+#' alignments by half the immuno-precipitated fragment size towards the
+#' middle of fragments. \code{shift} is either an \dQuote{integer} vector
+#' with one value per alignment file in \code{proj}, or a single
+#' \dQuote{integer} value, in which case all alignment files will be
+#' shifted by the same value. For paired-end experiments, it can be
+#' alternatively set to "halfInsert", which will estimate the true
+#' fragment size from the distance between aligned read pairs and shift
+#' the alignments accordingly.
+#' 
+#' \code{orientation} controls the interpretation of alignment strand
+#' when counting, relative to the strand of the query region. \code{any}
+#' will count all overlapping alignments, irrespective of the alignment
+#' strand (e.g. used in an unstranded RNA-seq experiment). \code{same}
+#' will only count the alignments on the same strand as the query region
+#' (e.g. in a stranded RNA-seq experiment), and \code{opposite} will only
```

Comment:
While we are at it: What about "(e.g. in a stranded RNA-seq experiment that generates reads from the sense strand)."

Reply:
Fixed